net/uds.go

```diff
@@ -36,6 +36,19 @@ func NewUDSListener(cfg *config.AgentConfig) (*UDSListener, error) {
 		return nil, fmt.Errorf("can't set the socket at write only: %s", err)
 	}
 
+	fileInfo, err := os.Stat(socketPath)
```

Comment:
Can we add tests around this logic? I think the main agent might have some similar tests that we can base ours on.